lib/engine/game/g_1848/step/dividend.rb

```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+
+require_relative '../../../step/dividend'
+
+module Engine
+  module Game
+    module G1848
+      module Step
+        class Dividend < Engine::Step::Dividend
+          BOE_BASE_PAYOUT = { '2' => 0, '3' => 100, '4' => 100, '5' => 200, '6' => 200, '8' => 300 }.freeze
+
+          def actions(entity)
+            return super unless entity == @game.boe
+
+            # boe always pays if it has any revenue
+            process_dividend(Action::Dividend.new(entity, kind: 'payout'))
```

Comment:
Do this as an auto_action. State should not change in the actions method.

Reply:
I rewrote it to this:
 def auto_actions(entity)
            return super unless entity == @game.boe
            [Action::Dividend.new(entity, kind: 'payout')]
          end
          
 and i'm getting an infinte loop. any ideas why? what am i missing?

Reply:
actions() is likely returning an action for BoE, which will trigger auto_actions to be called.

1) Is actions() gated by entity == current_entity?
2) After the dividend is processed, is the step passing?

Reply:
done, adding super after the action worked

Reply:
hmm, removing both actions and auto_actions, seems to still work. BOE pays out dividends when it has any, otherwise skips.
I have no idea why, thoughts?

Reply:
Likely the skip! method. That's how automatic dividends were done before auto_actions existed. See minor_half_pay.rb as an example.

Reply:
that's nasty. Yeah the skip! calls process_dividend with subtype withold, which i ignored. So regardless of wether process_dividend was called via skip or not, dividends were being paid.
Why would you name a method skip, if it still calls the procss_dividend action?

Reply:
fixed, added auto actions